src/c++/perf_analyzer/client_backend/triton/triton_client_backend.cc

```diff
@@ -362,6 +367,153 @@ TritonClientBackend::ModelInferenceStatistics(
   return Error::Success;
 }
 
+Error
+TritonClientBackend::TritonMetrics(
+    triton::perfanalyzer::TritonMetrics& triton_metrics)
+{
+  try {
+    std::string metrics_endpoint_text{""};
+    AccessTritonMetricsEndpoint(metrics_endpoint_text);
+    ParseAndStoreTritonMetrics(metrics_endpoint_text, triton_metrics);
+  }
+  catch (const std::exception& e) {
+    return Error(e.what(), pa::GENERIC_ERROR);
+  }
+  return Error::Success;
+}
+
+void
+TritonClientBackend::AccessTritonMetricsEndpoint(
+    std::string& metrics_endpoint_text)
+{
+  CURL* curl{curl_easy_init()};
+  if (curl == nullptr) {
+    throw std::runtime_error("Error calling curl_easy_init()");
```

Comment:
Let's use the PerfAnalyzerException that @dzier added. Returning different exceptions makes it unclear on what kind of handling is required for each function call.

Reply:
That's not merged into either main or the target branch for this PR.

Discussed offline and I catch this before the API returns to external library.

Reply:
Sorry for not being clear. I thought we wanted to return `Error` for now until David's PR is merged.

Reply:
It does eventually return an error here:

https://github.com/triton-inference-server/client/blob/matthewkotila-metric-collection/src/c%2B%2B/perf_analyzer/client_backend/triton/triton_client_backend.cc#L379